decoder/pokemonRtree.go

```diff
@@ -321,6 +329,23 @@ func GetPokemonInArea(retrieveParameters ApiPokemonScan) []*ApiPokemonResult {
 	pokemonSkipped := 0
 
 	isPokemonMatch := func(pokemonLookup *PokemonLookup, pvpLookup *PokemonPvpLookup, filter ApiPokemonFilter) bool {
+		if filter.Expert != nil {
```

Comment:
The lookup of the compiled expression (and all expression compilation if needed) could be moved outside of the loop.
The filter struct could have a 'filterExpr' added that was pre-populated so all that would need to be done is create the environment and run the compiled code.
Then any additional time would strictly be execution of the compiled filter.

Reply:
It's intentionally done this way so that we do not compile Pokemon filter that was not hit, for example, maybe there was a filter for Rayquaza but they do not spawn on map.

Reply:
Also not sure why the compilation time needs to be taken out. It is part of the overall round-trip time?

Reply:
In my tests I only looked at execution because I made the assumption that repeated calls would use similar filters (that would have been precompiled).
Which is why this was the only time I measured in my extended 10m loop for antlr and the equivelant expr code (which I did not push to github).